src/ingest-pipeline/misc/tools/zarr_zipper.py

```diff
@@ -0,0 +1,130 @@
+#! /usr/bin/env python
+
+import argparse
+import re
+import os
+import json
+from pathlib import Path
+from pprint import pprint
+from typing import List, Tuple, TypeVar, Union
+import logging
+from subprocess import run
+import requests
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+UUID_REGEX = re.compile(r"[0123456789aAbBcCdDeEfF]{32}")
+
+def parse_path(path) -> Tuple[Path, str, str]:
+    LOGGER.debug(f"starting with {path}")
+    target_path = path.parent
+    uuid_path = target_path
+    elts = list(uuid_path.parts)
+    LOGGER.debug(f"elts: {elts}")
+    elts.reverse()
+    for elt in elts:
+        LOGGER.debug(f"testing {elt}")
+        if UUID_REGEX.match(str(elt)):
+            uuid = str(elt)
+            break
+    else:
+        raise RuntimeError("no UUID found")
+    return target_path, str(path.name), uuid
+
+
+def run_zip(target_path: Path, zarr_name: str) -> None:
+    """
+    Zips the directory appropriately for use as a zipped zarr, and deletes
+    the original.
+
+    cd '/hive/hubmap-dev/data/consortium/IEC Testing Group/672111342c0002f90e1cb1bc5686ecf1/anndata-zarr/reg001_expr-anndata.zarr'
+    zip -r ../reg001_expr-anndata.zarr.zip .
+    cd ..
+    mv reg001_expr-anndata.zarr ../../672111342c0002f90e1cb1bc5686ecf1_anndata-zarr_reg001_expr-anndata.zarr
+    """
+    full_target_path = target_path / zarr_name
+    zip_name = target_path / f"{zarr_name}.zip"
+    cmd1 = f"cd '{full_target_path}'"
+    cmd2 = f"zip -r '{zip_name}' ."
+    cmd3 = "cd .."
+    cmd4 = f"rm -r -v {zarr_name}"
```

Comment:
Can we add a flag for the removal? IE, only remove if we set that flag specifically.